### PR TITLE
JitRegister: Use fmt

### PR DESCRIPTION
**Pull request by merryhime**
Prefer fmt to printf-style formatting.

### CODE REVIEW

File: Source/Core/Common/JitRegister.h [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 
 #pragma once
+#include <fmt/format.h>
 #include <stdarg.h>
 #include <string>
 #include "Common/CommonTypes.h"
```
Comment (Pokechu22) [inline Source/Core/Common/JitRegister.h:8]: Although this is fine, I think it'd be better to order it like this (and `stdarg.h` is no longer needed):

```C++
#pragma once

#include <string>

#include <fmt/format.h>

#include "Common/CommonTypes.h"
```

File: Source/Core/Common/JitRegister.h [hunk 2]
```diff
@@ -10,23 +11,22 @@ namespace JitRegister
...
+template <typename... Args>
+inline void Register(const void* base_address, u32 code_size, fmt::string_view format,
+                     const Args&... args)
 {
-  va_list args;
-  va_start(args, format);
-  RegisterV(base_address, code_size, format, args);
-  va_end(args);
+  RegisterV(base_address, code_size, format, fmt::make_format_args(args...));
 }
```
Comment (Pokechu22) [inline Source/Core/Common/JitRegister.h:23]: With slightly different syntax, it'd be possible to use `fmt`'s compile-time format checking, which only applies on C++20 but the MSVC builder targets C++20 so we'd still benefit from it:

```C++
template <typename... Args>
inline void Register(const void* base_address, u32 code_size, fmt::format_string<Args...> format,
                     Args&&... args)
{
  Register(base_address, code_size, fmt::format(format, std::forward<Args>(args)...));
}
```

The same would apply to the other `Register` function below.

**Review by Pokechu22 — COMMENTED** [review]
Nice idea.  I have some minor suggestions to enable `fmt`'s compile-time checking, but the changes otherwise look good.

**Comment by merryhime** [comment]
Sounds resonable to me. Updated.

**Comment by Pokechu22** [comment]
Looks good, but can you squash the two commits together?

**Review by Pokechu22 — APPROVED** [review]

**Pull request merged**